src/basho_bench_stats_writer_csv.erl

```diff
@@ -133,6 +172,53 @@ measurement_csv_file({Label, _Op}) ->
     ok = file:write(F, <<"elapsed, window, n, min, mean, median, 95th, 99th, 99_9th, max, errors\n">>),
     F.
 
+
+
+dump_run_statistics(RunStatsType) ->
+    Lines = stringify_stats(RunStatsType, erlang:get(run_metrics)),
+    TestDir = basho_bench:get_test_dir(),
+    FileName = filename:join([TestDir, "run_statistics." ++ atom_to_list(RunStatsType)]),
+    write_run_statistics(FileName, Lines).
+
+stringify_stats(_RunStatsType=json, RunMetrics) ->
+    [ jiffy:encode( {[{recordedMetrics, {RunMetrics}}]}, [pretty] ) ];
+stringify_stats(_RunStatsType=csv, RunMetrics) ->
+    %% Generate list of every known header
+    Headers = lists:foldl(fun(K, Acc) ->
+        {MetricsObj} = proplists:get_value(K, RunMetrics),
+        proplists:get_keys(MetricsObj) ++ Acc
+    end, [], proplists:get_keys(RunMetrics)),
+
+    %% Remove all duplicates and sort list of available headers, write to file
+    UniqueHeaders = lists:usort(Headers),
+    StringyfiedHeader =  io_lib:format("~s\n", 
+        [string:join( 
+           lists:map(fun atom_to_list/1, [op|UniqueHeaders]), ", ") ]),
```

Comment:
I would split this function out into separate constructs as it's a bit confusing in its current form. Maybe something like:

```
StrHeaders = string:join(lists:map(fun atom_to_list/1, [op|UniqueHeaders]), ", "),
StringyfiedHeader =  io_lib:format("~s\n", [StrHeaders]),
```